### ISSUE TITLE
Difference in the number of initialised modules on running terraform init

### ISSUE THREAD

**Issue by rohanpaithankar-theiconic** [terraform-google-modules/terraform-example-foundation#780]
### TL;DR

I ran the terraform init one week ago (03/08/2022) and it downloaded all the dependent modules. A few days later (11/08/2022), there are fewer modules being downloaded. No changes have been made to the scripts.

### Expected behavior

These are the old logs for modules expected to be downloaded on running `terraform init` command:

```
Already have image (with digest): {container registry path}
./verify_dependencies.sh
+ TERRAFORM_VERSION=v1.0.7
+ grep 'Terraform v1.0.7'
+ terraform version
Terraform v1.0.7
cp providers.template.tf  components/bootstrap/providers.tf
cd components/bootstrap && terraform init
Initializing modules...
Downloading terraform-google-modules/group/google 0.4.0 for billing_admins...
- billing_admins in .terraform/modules/billing_admins
- bootstrap in ../../modules/bootstrap
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.cloudbuild_bootstrap...
- bootstrap.cloudbuild_bootstrap in .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.cloudbuild_bootstrap.cloudbuild_project...
- bootstrap.cloudbuild_bootstrap.cloudbuild_project in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.budget in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/budget
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.gsuite_group in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/gsuite_group
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/core_project_factory
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory.project_services in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/project_services
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.shared_vpc_access in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/shared_vpc_access
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.seed_bootstrap...
- bootstrap.seed_bootstrap in .terraform/modules/bootstrap.seed_bootstrap
Downloading terraform-google-modules/kms/google 1.2.0 for bootstrap.seed_bootstrap.kms...
- bootstrap.seed_bootstrap.kms in .terraform/modules/bootstrap.seed_bootstrap.kms
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.seed_bootstrap.seed_project...
- bootstrap.seed_bootstrap.seed_project in .terraform/modules/bootstrap.seed_bootstrap.seed_project
- bootstrap.seed_bootstrap.seed_project.budget in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/budget
- bootstrap.seed_bootstrap.seed_project.gsuite_group in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/gsuite_group
- bootstrap.seed_bootstrap.seed_project.project-factory in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/core_project_factory
- bootstrap.seed_bootstrap.seed_project.project-factory.project_services in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/project_services
- bootstrap.seed_bootstrap.seed_project.shared_vpc_access in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/shared_vpc_access
Downloading terraform-google-modules/group/google 0.4.0 for organization_admins...
- organization_admins in .terraform/modules/organization_admins
Downloading terraform-google-modules/group/google 0.4.0 for security_admins...
- security_admins in .terraform/modules/security_admins

Initializing the backend...
```

### Observed behavior

Some root modules are not being downloaded, leading to 'Unsupported argument' errors in child modules on running `terraform plan` command. Following are the logs for modules being downloaded currently while runing the `terraform init` command.

```
Already have image (with digest): {container registry path}
./verify_dependencies.sh
+ TERRAFORM_VERSION=v1.0.7
+ terraform version
+ grep 'Terraform v1.0.7'
Terraform v1.0.7
cp providers.template.tf  components/bootstrap/providers.tf
cd components/bootstrap && terraform init
Initializing modules...
- bootstrap in ../../modules/bootstrap
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.cloudbuild_bootstrap...
- bootstrap.cloudbuild_bootstrap in .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.cloudbuild_bootstrap.cloudbuild_project...
- bootstrap.cloudbuild_bootstrap.cloudbuild_project in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project
Downloading terraform-google-modules/bootstrap/google 3.1.0 for bootstrap.seed_bootstrap...
- bootstrap.seed_bootstrap in .terraform/modules/bootstrap.seed_bootstrap
Downloading terraform-google-modules/kms/google 1.2.0 for bootstrap.seed_bootstrap.kms...
- bootstrap.seed_bootstrap.kms in .terraform/modules/bootstrap.seed_bootstrap.kms
Downloading terraform-google-modules/project-factory/google 10.1.1 for bootstrap.seed_bootstrap.seed_project...
- bootstrap.seed_bootstrap.seed_project in .terraform/modules/bootstrap.seed_bootstrap.seed_project
- Downloading terraform-google-modules/group/google 0.4.0 for organization_admins...
- organization_admins in .terraform/modules/organization_admins
Downloading terraform-google-modules/group/google 0.4.0 for security_admins...
- security_admins in .terraform/modules/security_admins

Initializing the backend...
```

The following logs are not present:
```
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.budget in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/budget
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.gsuite_group in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/gsuite_group
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/core_project_factory
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.project-factory.project_services in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/project_services
- bootstrap.cloudbuild_bootstrap.cloudbuild_project.shared_vpc_access in .terraform/modules/bootstrap.cloudbuild_bootstrap.cloudbuild_project/modules/shared_vpc_access
- bootstrap.seed_bootstrap.seed_project.budget in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/budget
- bootstrap.seed_bootstrap.seed_project.gsuite_group in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/gsuite_group
- bootstrap.seed_bootstrap.seed_project.project-factory in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/core_project_factory
- bootstrap.seed_bootstrap.seed_project.project-factory.project_services in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/project_services
- bootstrap.seed_bootstrap.seed_project.shared_vpc_access in .terraform/modules/bootstrap.seed_bootstrap.seed_project/modules/shared_vpc_access
```

### Terraform Configuration

```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = ">= 3.50"
    }
  }
}
```

### Terraform Version

```sh
Terraform v1.0.7
```

**Comment by bharathkkb** [comment]
Could you also add the `Unsupported argument` errors being thrown?

**Comment by rohanpaithankar-theiconic** [comment]
Here are the errors as per the logs:

```
Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 43, in module "cloudbuild_project":
│   43:   name                        = local.cloudbuild_project_id
│ 
│ An argument named "name" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 44, in module "cloudbuild_project":
│   44:   random_project_id           = var.random_suffix
│ 
│ An argument named "random_project_id" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 45, in module "cloudbuild_project":
│   45:   disable_services_on_destroy = false
│ 
│ An argument named "disable_services_on_destroy" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 46, in module "cloudbuild_project":
│   46:   folder_id                   = var.folder_id
│ 
│ An argument named "folder_id" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 47, in module "cloudbuild_project":
│   47:   org_id                      = var.org_id
│ 
│ An argument named "org_id" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 48, in module "cloudbuild_project":
│   48:   billing_account             = var.billing_account
│ 
│ An argument named "billing_account" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 49, in module "cloudbuild_project":
│   49:   activate_apis               = local.activate_apis
│ 
│ An argument named "activate_apis" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/bootstrap.cloudbuild_bootstrap/modules/cloudbuild/main.tf line 50, in module "cloudbuild_project":
│   50:   labels                      = var.project_labels
│ 
│ An argument named "labels" is not expected here.
╵
```
